webserver_configurations/OpenID_Connect/Nginx/conf.d/second_opinion/openidc_second_opinion_layer.lua

```diff
@@ -0,0 +1,176 @@
+-- Lua reference for nginx: https://github.com/openresty/lua-nginx-module
+-- Lua reference for openidc: https://github.com/pingidentity/lua-resty-openidc
+local oidc = require("resty.openidc")
+local cjson = require( "cjson" )
+
+Set = {}
+
+function Set.new (t)
+  local set = {}
+  for _, l in ipairs(t) do set[l] = true end
+  return set
+end
+
+function Set.union (a,b)
+  local res = Set.new{}
+  for k in pairs(a) do res[k] = true end
+  for k in pairs(b) do res[k] = true end
+  return res
+end
+
+function Set.intersection (a,b)
+  local res = Set.new{}
+  for k in pairs(a) do
+    res[k] = b[k]
+  end
+  return res
+end
+
+function Set.a_not_in_b (a,b)
+  local res = Set.new{}
+  for k in pairs(a) do
+    if b[k] == nil then
+      res[k] = a[k]
+    end
+  end
+  return res
+end
+
+function Set.tostring (set)
+  local s = "{"
+  local sep = ""
+  for e in pairs(set) do
+    s = s .. sep .. e
+    sep = ", "
+  end
+  return s .. "}"
+end
+
+function Set.print (s)
+  print(Set.tostring(s))
+end
+
+function Set.totable(set)
+  local res={}
+  local n=0
+
+  for k,_ in pairs(set) do
+    n=n+1
+    res[n]=k
+  end
+  return res
+end
+
+local function table_to_string(table)
+  local s = ""
+  for _,v in pairs(table) do
+    s = s and s.."|"..v or v
+  end
+  return s
+end
+
+local function compare_sessions(s1, s2)
+  -- check that both OPs authenticated the same user
+  local res = s1.data.user.email == s2.data.user.email
+  if res == false then
+    ngx.log(ngx.ERR, "User authenticated as different identities in primary and secondary auth")
+  end
+  return res
+end
+
```

Comment:
@gdestuynder This section checks to validate that the user which authenticated to each OP is the same user (comment is `check that both OPs authenticated the same user`)

Was this note just meant for the section of the code that finds the intersection of the `groups` returned by each OP?